nb-engine/nb-engine-core/src/main/java/io/nosqlbench/engine/api/scenarios/NBCLIScenarioPreprocessor.java

```diff
@@ -252,10 +252,66 @@ public static String sanitize(String word) {
 
     private static final Pattern WordAndMaybeAssignment = Pattern.compile("(?<name>\\w[-_\\d\\w.]+)((?<oper>=+)(?<val>.+))?");
 
+    public static List<String> splitCommand(String cmd) {
```

Comment:
@Prodyte It is possible to avoid spaces inside quotes using this regex `s+(?=([^"]*"[^"]*")*[^"]*$)` inside split method
Still the quotes will remain
```
# Running directly from commandline
$ java -jar nb5.jar run workload=iot.yaml tags=block:main-read cycles=10 threads=1 driver=stdout param="test 1"
test 1
test 1
test 1
...

# Running from scenario (after patching with above regex)
$ head -n 8 iot.yaml | tail -n 3
scenarios:
  test:
    read: run driver=stdout tags==block:main-read cycles==10 threads=auto param="test 1"
$ java -jar nb5.jar iot.yaml test
"test 1"
"test 1"
"test 1"
...
```
So the quotes remained, In case we actually need the quotes in the param string we can use escape character to include it
Thats how you can add quotes to the params when running directly from command line

One other thing that I thought is that we can replace these quotes and handle escape characters using regex
But then I don't think we can check if the quotes are balanced or not using regex or any other one liner code

Let me know if you can think of cleaner way to handle that too

Reply:
@Prodyte It is possible to avoid spaces inside quotes using this regex `s+(?=([^"]*"[^"]*")*[^"]*$)` inside split method
Still the quotes will remain
```
# Running directly from commandline
$ java -jar nb5.jar run workload=iot.yaml tags=block:main-read cycles=10 threads=1 driver=stdout param="test 1"
test 1
test 1
test 1
...

# Running from scenario (after patching with above regex)
$ head -n 8 iot.yaml | tail -n 3
scenarios:
  test:
    read: run driver=stdout tags==block:main-read cycles==10 threads=auto param="test 1"
$ java -jar nb5.jar iot.yaml test
"test 1"
"test 1"
"test 1"
...
```
So the quotes remained, In case we actually need the quotes in the param string we can use escape character to include it
Thats how you can add quotes to the params when running directly from command line

One other thing that I thought is that we can replace these quotes and handle escape characters using regex
But then I don't think we can check if the quotes are balanced or not using regex or any other one liner code

Let me know if you can think of cleaner way to handle that too

Reply:
For checking balanced quotes you can create a method, for everything else you can use regex.